htdocs/class/xoopssecurity.php

```
@@ -261,10 +261,24 @@ public function checkBadips()
+            // and treat any non-match — including a compile failure — as "allowed",
+            // so a bad pattern neither warns nor blocks the request (SECURITY.md L-19).
+            set_error_handler(static fn (): bool => true);
+            try {
+                foreach ($xoopsConfig['bad_ips'] as $bi) {
+                    // Bound the length to avoid pathological backtracking and block
+                    // only on a definite match.
+                    if (!is_string($bi) || '' === $bi || strlen($bi) > 255) {
+                        continue;
+                    }
+                    if (1 === preg_match('/' . $bi . '/', $ip)) {
+                        exit();
```

PHPMD (2.15.0)

[error] The method checkBadips() contains an exit expression. (undefined)

(ExitExpression)